src/Component/Metadata/MetadataInterface.php

```diff
@@ -13,6 +13,9 @@
 
 namespace Sylius\Component\Resource\Metadata;
 
+/**
+ * @method string|null getStateMachineComponent()
```

Comment:
Do we need this soft method requirement?

Reply:
Yeah I think that's better to inform that this method has been introduced.

Reply:
Hum that let me think that I have to check if this method exists when I use it on a service.
Hum.. What do you think?

Reply:
either new interface or always expect this method to be present